src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala

```diff
@@ -0,0 +1,349 @@
+      source: DescriptorProto,
+      files: List[FileDescriptorProto]): NativeFieldF =
+    (field.getLabel.isRepeated, isMap(field, source)) match {
+      case (true, false) =>
+        NativeField(
+          name = field.getName,
+          position = field.getNumber,
+          tpe = NativeRepeated(fromFieldType(field, files)),
+          options = fromFieldOptionsMsg(field.getOptions),
+          isRepeated = true,
+          isMapField = false
+        )
+      case (_, true) =>
+        NativeField(
+          name = field.getName,
+          position = field.getNumber,
+          tpe = getNativeMap(field.getName, source, files),
+          options = fromFieldOptionsMsg(field.getOptions),
+          isRepeated = false,
+          isMapField = true
+        )
+      case _ =>
+        NativeField(
+          name = field.getName,
+          position = field.getNumber,
+          tpe = fromFieldType(field, files),
+          options = fromFieldOptionsMsg(field.getOptions),
+          isRepeated = field.getLabel.isRepeated,
+          isMapField = isMap(field, source)
+        )
+    }
+
+  def isMap(field: FieldDescriptorProto, source: DescriptorProto): Boolean =
+    source.getNestedTypeList.j2s.exists(
+      e =>
+        e.getOptions.getMapEntry &&
+          matchNameEntry(field.getName, e) &&
+          takeOnlyMapEntries(e.getFieldList.j2s))
+
+  def getNativeMap(name: String, source: DescriptorProto, files: List[FileDescriptorProto]): NativeMap =
+    (for {
+      maybeMsg <- source.getNestedTypeList.j2s
+        .find(e => e.getOptions.getMapEntry && matchNameEntry(name, e) && takeOnlyMapEntries(e.getFieldList.j2s))
+      maybeKey   <- getMapField(maybeMsg, "key")
+      maybeValue <- getMapField(maybeMsg, "value")
+    } yield NativeMap(fromFieldType(maybeKey, files), fromFieldType(maybeValue, files)))
+      .getOrElse(throw new Exception(s"Could not find map entry for: $name"))
+
+  def getMapField(msg: DescriptorProto, name: String): Option[FieldDescriptorProto] =
+    msg.getFieldList.j2s.find(_.getName == name)
+
+  def takeOnlyMapEntries(fields: List[FieldDescriptorProto]): Boolean =
+    fields.count(f => (f.getNumber == 1 && f.getName == "key") || (f.getNumber == 2 && f.getName == "value")) == 2
+
+  def matchNameEntry(name: String, source: DescriptorProto): Boolean =
+    source.getName.toLowerCase == s"${name}Entry".toLowerCase
+
+  def fromOneofDescriptorsProto(
+      oneOfFields: List[OneofDescriptorProto],
+      fields: List[FieldDescriptorProto],
+      source: DescriptorProto,
+      files: List[FileDescriptorProto]): List[NativeFieldF] = oneOfFields.zipWithIndex.map {
+    case (oneof, index) => {
+      val oneOfFields: NonEmptyList[NativeField] = NonEmptyList
+        .fromList(
+          fields
+            .filter(t => t.hasOneofIndex && t.getOneofIndex == index)
+            .map(fromFieldDescriptorProto(_, source, files))
+            .collect { case b: NativeField => b })
```

Comment:
I believe this `b: NativeField` is doing a `ClassTag` check at runtime.  Can you change it with a pattern match like:

```suggestion
            .collect { case b@NativeField(_, _, _, _, _, _) => b })
```

Reply:
Solved at https://github.com/higherkindness/skeuomorph/pull/66/commits/78581e098293b077fa76b8987819e49fba0d4703